### PR TITLE
[ci] Artifact uploading before test in release.yml

### PR DESCRIPTION
**Pull request by Leonz5288**
Related issue = #

### CODE REVIEW

**Review by ailzhang — APPROVED** [review]
Awesome! Thanks!

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -136,6 +142,12 @@ jobs:
           CI_SETUP_CMAKE_ARGS: -DTI_WITH_VULKAN:BOOL=OFF -DTI_WITH_OPENGL:BOOL=OFF -DTI_WITH_CC:BOOL=OFF -DTI_BUILD_TESTS:BOOL=${{ matrix.with_cpp_tests }}
           PROJECT_NAME: ${{ matrix.name }}
 
+      - name: Archive Wheel Artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ matrix.name }}-py${{ matrix.python }}-macOS.whl
```
Comment (k-ye) [inline .github/workflows/release.yml:148]:
```suggestion
          name: ${{ matrix.name }}-py${{ matrix.python }}-macos.whl
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -190,6 +202,12 @@ jobs:
           PROJECT_NAME: ${{ matrix.name }}
           PYTHON: ${{ matrix.python }}
 
+      - name: Archive Wheel Artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ matrix.name }}-py${{ matrix.python }}-macM1.whl
```
Comment (k-ye) [inline .github/workflows/release.yml:208]:
```suggestion
          name: ${{ matrix.name }}-py${{ matrix.python }}-macos-m1.whl
```

**Review by k-ye — APPROVED** [review]
LGTM + super nits!

**Pull request merged**